src/simulators/matrix_product_state/matrix_product_state.hpp

```diff
@@ -663,6 +666,54 @@ void State::apply_gate(const Operations::Op &op) {
   }
 }
 
+  void State::apply_kraus(const reg_t &qubits,
+                   const std::vector<cmatrix_t> &kmats,
+                   RngEngine &rng) {
+  // Check edge case for empty Kraus set (this shouldn't happen)
+  if (kmats.empty())
+    return; // end function early
+
+  // Choose a real in [0, 1) to choose the applied kraus operator once
+  // the accumulated probability is greater than r.
+  // We know that the Kraus noise must be normalized
+  // So we only compute probabilities for the first N-1 kraus operators
+  // and infer the probability of the last one from 1 - sum of the previous
+
+  double r = rng.rand(0., 1.);
+  double accum = 0.;
+  bool complete = false;
+
+  // it is more efficient to compute the density matrix once rather than computing
+  // the expectation value in every iteration of the loop
+  cmatrix_t rho = qreg_.density_matrix(qubits);
```

Comment:
Unrelated to this PR, but if MPS supports computing the density matrix, we should add density matrix snapshot support to the class like we do for the statevector method.

Reply:
This could also help with testing Kraus noise is being applied correctly

Reply:
I will open a separate PR to support snapshot_density_matrix in MPS.